plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -639,35 +667,74 @@ private void addSplitPluginDependencies(String thisPlugin, MavenRunner.Config mc
+                    replacing.put(plugin, bundledV);
+                }
+                // Also check any dependencies, so if we are upgrading cloudbees-folder, we also add an explicit dep on a bundled credentials.
+                for (Map.Entry<String,String> dependency : bundledP.dependencies.entrySet()) {
+                    String depPlugin = dependency.getKey();
+                    if (pluginList.containsKey(depPlugin)) {
+                        continue; // already handled
                     }
-                    // Also check any dependencies, so if we are upgrading cloudbees-folder, we also add an explicit dep on a bundled credentials.
-                    for (Map.Entry<String,String> dependency : bundledP.dependencies.entrySet()) {
-                        String depPlugin = dependency.getKey();
-                        if (pluginDeps.containsKey(depPlugin)) {
-                            continue; // already handled
-                        }
-                        // We ignore the declared dependency version and go with the bundled version:
-                        Plugin depBundledP = otherPlugins.get(depPlugin);
-                        if (depBundledP != null) {
-                            System.out.println("Adding " + depPlugin + " since it was a dependency of " + plugin);
-                            toAdd.put(depPlugin, new VersionNumber(depBundledP.version));
-                        }
+                    // We ignore the declared dependency version and go with the bundled version:
+                    Plugin depBundledP = otherPlugins.get(depPlugin);
+                    if (depBundledP != null) {
+                        updateAllDependents(plugin, depBundledP, pluginList, adding, replacing, otherPlugins, inTest, toConvertFromTest);
                     }
                 }
             }
-            if (!toAdd.isEmpty() || !toReplace.isEmpty()) {
-                System.out.println("Adding/replacing plugin dependencies for compatibility: " + toAdd + " " + toReplace);
-                pom.addDependencies(toAdd, toReplace, coreDep, pluginGroupIds);
-            }
         }
     }
 
+    /**
+     * Search the dependents of a given plugin to determine if they need to be converted.
+     */
+    private void updateAllDependents(String parent, Plugin dependent, Map<String,VersionNumber> pluginList, Map<String,VersionNumber> adding, Map<String,VersionNumber> replacing, Map<String,Plugin> otherPlugins, List<String> inTest, List<String> toConvertFromTest) {
+        // Check if this exists with an undesired scope
+        String pluginName = dependent.name;
+        if(inTest.contains(pluginName)) { // becomes a replacement 
+            System.out.println("Converting " + pluginName + " from the test scope since it was a dependency of " + parent);
+            toConvertFromTest.add(pluginName);
+            replacing.put(pluginName, new VersionNumber(dependent.version));
```

Comment:
Sure.  The first one I can think of off the top of my head is the copyartifact plugin.  I can add a reference to it in a comment.